test/model.jl

```diff
@@ -328,6 +328,21 @@ function test_model()
         @test optimizer.b == 2
     end
 
+    @testset "solve_time" begin
+        @testset "NoOptimizer()" begin
+            err = NoOptimizer()
+            model = Model()
+            @test_throws err solve_time(model)
+        end
+
+        @testset "OptimizeNotCalled()" begin
+            err = OptimizeNotCalled()
+            model = Model(with_optimizer(MOIU.MockOptimizer,
+                                         SimpleLPModel{Float64}()))
+            @test_throws err solve_time(model)
+        end
```

Comment:
You can add a test where it works by using a UniversalFallback with mock opt

Reply:
I'm not getting exactly how to do this. I have tried
```julia
model = Model(with_optimizer(MOIU.MockOptimizer,
                             MOIU.UniversalFallback(JuMP._MOIModel{Float64}())))
JuMP.solve_time(model)
```
but gives an error
```julia
ERROR: OptimizeNotCalled()
Stacktrace:
 [1] _moi_get_result(::MathOptInterface.Utilities.CachingOptimizer{MathOptInterface.AbstractOptimizer,MathOptInterface.Utilities.UniversalFallback{JuMP._MOIModel{Float64}}}, ::MathOptInterface.SolveTime) at /Users/guilhermebodin/Documents/JuMP.jl/src/JuMP.jl:594
 [2] get(::Model, ::MathOptInterface.SolveTime) at /Users/guilhermebodin/Documents/JuMP.jl/src/JuMP.jl:609
 [3] solve_time(::Model) at /Users/guilhermebodin/Documents/JuMP.jl/src/JuMP.jl:448
 [4] top-level scope at none:0
```
and 

```julia
model = Model(with_optimizer(MOIU.MockOptimizer,
                             MOIU.UniversalFallback(JuMP._MOIModel{Float64}())))   

optimize!(model)
JuMP.solve_time(model)
```
gives the same error
```julia
ERROR: OptimizeNotCalled()
Stacktrace:
 [1] _moi_get_result(::MathOptInterface.Utilities.CachingOptimizer{MathOptInterface.AbstractOptimizer,MathOptInterface.Utilities.UniversalFallback{JuMP._MOIModel{Float64}}}, ::MathOptInterface.SolveTime) at /Users/guilhermebodin/Documents/JuMP.jl/src/JuMP.jl:594
 [2] get(::Model, ::MathOptInterface.SolveTime) at /Users/guilhermebodin/Documents/JuMP.jl/src/JuMP.jl:609
 [3] solve_time(::Model) at /Users/guilhermebodin/Documents/JuMP.jl/src/JuMP.jl:448
 [4] top-level scope at none:0
```

Reply:
You can try
```julia
mock = MOIU.MockOptimizer(MOIU.UniversalFallback(JuMP._MOIModel{Float64}())))
mock.optimize! = mock -> begin
    MOI.set(mock, MOI.TerminationStatus(), MOI.OPTIMAL)
    MOI.set(mock, MOI.SolveTime(), 2.0)
end
model = direct_model(mock)
optimize!(model)
@test solve_time(model) == 2.0
```

Reply:
This gives this error
```julia
ERROR: ArgumentError: `supports` is not defined for MathOptInterface.SolveTime(), it is only defined for attributes such that `is_copyable` returns `true`.
Stacktrace:
 [1] supports(::JuMP._MOIModel{Float64}, ::MathOptInterface.SolveTime) at /Users/guilhermebodin/.julia/packages/MathOptInterface/C3lip/src/attributes.jl:118
 [2] set(::MathOptInterface.Utilities.UniversalFallback{JuMP._MOIModel{Float64}}, ::MathOptInterface.SolveTime, ::Float64) at /Users/guilhermebodin/.julia/packages/MathOptInterface/C3lip/src/Utilities/universalfallback.jl:284
 [3] set(::MathOptInterface.Utilities.MockOptimizer{MathOptInterface.Utilities.UniversalFallback{JuMP._MOIModel{Float64}}}, ::MathOptInterface.SolveTime, ::Float64) at /Users/guilhermebodin/.julia/packages/MathOptInterface/C3lip/src/Utilities/mockoptimizer.jl:157
 [4] (::getfield(Main, Symbol("##17#18")))(::MathOptInterface.Utilities.MockOptimizer{MathOptInterface.Utilities.UniversalFallback{JuMP._MOIModel{Float64}}}) at ./REPL[61]:3
 [5] optimize!(::MathOptInterface.Utilities.MockOptimizer{MathOptInterface.Utilities.UniversalFallback{JuMP._MOIModel{Float64}}}) at /Users/guilhermebodin/.julia/packages/MathOptInterface/C3lip/src/Utilities/mockoptimizer.jl:113
 [6] #optimize!#79(::Bool, ::Bool, ::Function, ::Model, ::Nothing) at /Users/guilhermebodin/Documents/JuMP.jl/src/optimizer_interface.jl:132
 [7] optimize! at /Users/guilhermebodin/Documents/JuMP.jl/src/optimizer_interface.jl:105 [inlined] (repeats 2 times)
 [8] top-level scope at none:0
```
because of the MOI function 
```julia
"""
    is_set_by_optimize(::AnyAttribute)

Return a `Bool` indicating whether the value of the attribute is modified
during an [`optimize!`](@ref) call, that is, the attribute is used to query
the result of the optimization.

## Important note when defining new attributes

This function returns `false` by default so it should be implemented for
attributes that are modified by [`optimize!`](@ref).
"""
is_set_by_optimize(::AnyAttribute) = false
function is_set_by_optimize(::Union{ObjectiveValue,
                                    ObjectiveBound,
                                    RelativeGap,
                                    SolveTime,
                                    SimplexIterations,
                                    BarrierIterations,
                                    NodeCount,
                                    RawSolver,
                                    ResultCount,
                                    TerminationStatus,
                                    RawStatusString,
                                    PrimalStatus,
                                    DualStatus,
                                    VariablePrimal,
                                    ConstraintPrimal,
                                    ConstraintDual,
                                    ConstraintBasisStatus})
    return true
end
```
is there another way to call the optimizer, I mean without having to call the optimize! function?

Reply:
Indeed, you can leave a TODO while https://github.com/JuliaOpt/MathOptInterface.jl/issues/707 is not solved